src/analysis/subadditive.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2021 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import topology.instances.real
+import order.filter.archimedean
+
+/-!
+# Convergence of subadditive sequences
+
+A subadditive sequence `u : ℕ → ℝ` is a sequence satisfying `u (m + n) ≤ u m + u n` for all `m, n`.
+We define this notion as `subadditive u`, and prove in `subadditive.tendsto_lim` that, if `u n / n`
+is bounded below, then it converges to a limit (that we denote by `subadditive.lim` for
+convenience). This result is known as Fekete's lemma in the literature.
+-/
+
+noncomputable theory
+open set filter
+open_locale topological_space
+
+/-- A real-valued sequence is subadditive if it satisfies the inequality `u (m + n) ≤ u m + u n`
+for all `m, n`. -/
+def subadditive (u : ℕ → ℝ) : Prop :=
```

Comment:
should we generalize the type of `u` (at least in this definition, but maybe also for the proof)?

Reply:
I had thought a little bit about it, but I did not find a convincing typeclass. You need a conditionally complete linear order and a compatible field structure, so reals are basically the only example (and we have no general typeclass where this would make sense).